### PR TITLE
Generate XPLINK call descriptor inline to handle large methods

### PR DESCRIPTION
**Pull request by fjeremic**
We generate the XPLINK call descriptor inline right after the call
instead of the literal pool because some z/OS 31-bit programs tend to
be rather large, and the distance between the call location and the
literal pool may exceed the number of bits we have to encode the offset
in the NOP descriptor.

Signed-off-by: Filip Jeremic <fjeremic@ca.ibm.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
I've generated traces with the call instruction occurring at all possible alignments to test what gets generated. By manually examining the call descriptor location I have convinced myself the change is correct. I have also ran the change on z/OS 31-bit OpenJ9 testing which exercises calls to non-XPLINK code (see eclipse/openj9#6614). All tests are passing.

Requesting @joransiu and @klangman for review.

**Comment by fjeremic** [comment]
Here are the traces with all 4 possibilities of alignment (0, 2, 4, 6):

```
0x49665F60 0000046c [0x4B53F9D0] 0d 76                      BASR    GPR7,GPR6
0x49665F62 0000046e [0x4B53FD30]                            Label L0257:
0x49665F62 0000046e [0x4B53FE30] 47 00 00 02                DC      0x47000000
0x49665F66 00000472 [0x4B53FED0] a7 f4 00 09                BRC     NOP(0xf), Label L0273, labelTargetAddr=0x49665F78
0x49665F6A 00000476 [0x4B752850] c0 04 00 00 00 00          NOP
0x49665F70 0000047c [0x4B53FFD0]                            Label L0272:
0x49665F70 0000047c [0x4B540050] 00 00 00 00                DC      0x00000000
0x49665F74 00000480 [0x4B5400D0] 01 00 00 00                DC      0x01000000
0x49665F78 00000484 [0x4B540440]                            ASSOCREGS
0x49665F78 00000484 [0x4B540150]                            Label L0273:

0x49646182 0000008a [0x4B2FE9C0] 0d 76                      BASR    GPR7,GPR6
0x49646184 0000008c [0x4B2FED20]                            Label L0048:
0x49646184 0000008c [0x4B2FEE20] 47 00 00 02                DC      0x47000000
0x49646188 00000090 [0x4B2FEEC0] a7 f4 00 08                BRC     NOP(0xf), Label L0052, labelTargetAddr=0x49646198
0x4964618C 00000094 [0x4B3B8670] 47 00 00 00                NOP
0x49646190 00000098 [0x4B2FEFC0]                            Label L0051:
0x49646190 00000098 [0x4B2FF040] 00 00 00 00                DC      0x00000000
0x49646194 0000009c [0x4B2FF0C0] 01 00 00 00                DC      0x01000000
0x49646198 000000a0 [0x4B2FF430]                            ASSOCREGS
0x49646198 000000a0 [0x4B2FF140]                            Label L0052:

0x4963E094 0000009c [0x4B312B80] 0d 76                      BASR    GPR7,GPR6
0x4963E096 0000009e [0x4B312EE0]                            Label L0032:
0x4963E096 0000009e [0x4B312FE0] 47 00 00 02                DC      0x47000000
0x4963E09A 000000a2 [0x4B313080] a7 f4 00 07                BRC     NOP(0xf), Label L0040, labelTargetAddr=0x4963E0A8
0x4963E09E 000000a6 [0x4B3CF370] 18 00                      NOP
0x4963E0A0 000000a8 [0x4B313180]                            Label L0039:
0x4963E0A0 000000a8 [0x4B313200] 00 00 00 00                DC      0x00000000
0x4963E0A4 000000ac [0x4B313280] 01 00 00 00                DC      0x01000000
0x4963E0A8 000000b0 [0x4B3135F0]                            ASSOCREGS
0x4963E0A8 000000b0 [0x4B313300]                            Label L0040:

0x4963EE96 00000082 [0x4B2EE500] 0d 76                      BASR    GPR7,GPR6
0x4963EE98 00000084 [0x4B2EE860]                            Label L0032:
0x4963EE98 00000084 [0x4B2EE960] 47 00 00 01                DC      0x47000000
0x4963EE9C 00000088 [0x4B2EEA00] a7 f4 00 06                BRC     NOP(0xf), Label L0038, labelTargetAddr=0x4963EEA8
0x4963EEA0 0000008c [0x4B2EEB00]                            Label L0037:
0x4963EEA0 0000008c [0x4B2EEB80] 00 00 00 00                DC      0x00000000
0x4963EEA4 00000090 [0x4B2EEC00] 01 00 00 00                DC      0x01000000
0x4963EEA8 00000094 [0x4B2EEF70]                            ASSOCREGS
0x4963EEA8 00000094 [0x4B2EEC80]                            Label L0038:
```

File: compiler/z/CMakeLists.txt [hunk 1]
```diff
@@ -57,7 +57,6 @@ compiler_library(z
 	${CMAKE_CURRENT_LIST_DIR}/codegen/OMRCodeGenerator.cpp
 	${CMAKE_CURRENT_LIST_DIR}/codegen/snippet/PPA1Snippet.cpp
 	${CMAKE_CURRENT_LIST_DIR}/codegen/snippet/PPA2Snippet.cpp
-	${CMAKE_CURRENT_LIST_DIR}/codegen/snippet/XPLINKCallDescriptorSnippet.cpp
```
Comment (aviansie-ben) [inline compiler/z/CMakeLists.txt]: Is there any particular reason behind not deleting the `XPLINKCallDescriptorSnippet.cpp` and `XPLINKCallDescriptorSnippet.hpp` files themselves? It looks like they're unused after this change.

Comment (fjeremic) [reply compiler/z/CMakeLists.txt]: Right, I should have just said it explicitly in a PR comment. The commit body (aa4d2cf) explains that we cannot remove this until downstream projects stop compiling this file.

Comment (fjeremic) [reply compiler/z/CMakeLists.txt]: Opened up eclipse/openj9#8739 to address this.

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by joransiu** [comment]
Just for reference, "the call offset in question is a 16-bit field containing the offset in doublewords from the call site to, if negative the entry point marker for the function or, if positive, a call descriptor, described below, which contains both the offset to the entry point marker and information about parameter and return types." [1]  That would give ~ +256k (minus alignment adjustments) distance to the call descriptor.   Java methods (call -> litpool distance) could be bigger than that.

[1] https://www.ibm.com/support/knowledgecenter/SSLTBW_2.2.0/com.ibm.zos.v2r2.ceev100/cee1v283.htm

**Comment by joransiu** [comment]
@fjeremic, your changes LGTM.  Can you just confirm that the alignment constraints will be respected on AOT load?  I expect that to be the case, given it is simply doubleword alignment requirement for the call descriptor, but wanted to confirm.

**Comment by fjeremic** [comment]
@joransiu just verified by filling in the cache via:

```
bin/java -XXjitdirectory=`pwd` -Xshareclasses:name=xplink "-Xaot:count=1,disableAsyncCompilation,verbose" "-Xjit:count=1,disableAsyncCompilation,verbose" -version
```

And the cache is now full (output of `printStats`):

```
>bin/java -XXjitdirectory=`pwd` -Xshareclasses:name=xplink,printStats -version
JVMJ9VM082E Unable to switch to IFA processor - issue "extattr +a libj9ifa29.so"

Current statistics for cache "xplink":

shared memory ID   = 204806
Cache created with:
        -Xnolinenumbers                      = false
        BCI Enabled                          = true
        Restrict Classpaths                  = false
        Feature                              = default

Cache contains only classes with line numbers

base address                         = 0x4B11C000
end address                          = 0x4C100000
allocation pointer                   = 0x4B26F680

cache layer                          = 0
cache size                           = 16776812
softmx bytes                         = 16776812
free bytes                           = 11827656
Reserved space for AOT bytes         = -1
Maximum space for AOT bytes          = -1
Reserved space for JIT data bytes    = -1
Maximum space for JIT data bytes     = -1
Metadata bytes                       = 73689
Metadata % used                      = 1%
Class debug area size                = 1331200
Class debug area used bytes          = 191618
Class debug area % used              = 14%

ROMClass bytes                       = 1390208
AOT bytes                            = 1185355
JIT data bytes                       = 34068
Zip cache bytes                      = 820352
Startup hint bytes                   = 0
Data bytes                           = 114284

# ROMClasses                         = 456
# AOT Methods                        = 870
# Classpaths                         = 1
# URLs                               = 0
# Tokens                             = 0
# Zip caches                         = 22
# Startup hints                      = 0
# Stale classes                      = 0
% Stale classes                      = 0%


Cache is 29% full

Cache is accessible to current user = true
```

There are 870 AOT compiled methods. I then ran `java -version` again to load all the AOT bodies, and generated a core dump on the `vmStop` event. I then opened the core dump in KCA and dumped the assembly of the entire JIT code cache. Then I grepped for "BASR      GPR7,GPR6" and found samples at each alignment and also validated that each location was an AOT load in the verbose log and the address range matches what we see in the verbose log.

Here are the results:

```

+ (AOT load) java/lang/Class.getName()Ljava/lang/String; @ 0x49622D4C-0x49622FEC Q_SZ=0 Q_SZI=0 QW=1 j9m=49568A94 bcsz=26 compThread=0

0x49622e82 {java/lang/Class.getName} +310          0D76         BASR      GPR7,GPR6
0x49622e84 {java/lang/Class.getName} +312          47000002     NOP       0x2
0x49622e88 {java/lang/Class.getName} +316          A7F40008     J         *0x10 U>> +332
0x49622e8c {java/lang/Class.getName} +320          47000000     NOP       0x0 <<< *Unreachable*
0x49622e90 {java/lang/Class.getName} +324          0000         Invalid Instruction
0x49622e92 {java/lang/Class.getName} +326          0000         Invalid Instruction
0x49622e94 {java/lang/Class.getName} +328          0100         Invalid Instruction
0x49622e96 {java/lang/Class.getName} +330          0000         Invalid Instruction
0x49622e98 {java/lang/Class.getName} +332          5040D51C     ST        GPR4,0x51C(,GPR13) <<< +316

+ (AOT load) java/io/UnixFileSystem.canonicalize(Ljava/lang/String;)Ljava/lang/String; @ 0x4962D0E8-0x4963111C Q_SZ=0 Q_SZI=0 QW=1 j9m=4C30B594 bcsz=238 compThread=0

0x4962d314 {java/io/UnixFileSystem.canonicalize} +556          0D76         BASR      GPR7,GPR6
0x4962d316 {java/io/UnixFileSystem.canonicalize} +558          47000002     NOP       0x2
0x4962d31a {java/io/UnixFileSystem.canonicalize} +562          A7F40007     J         *0xE U>> +576
0x4962d31e {java/io/UnixFileSystem.canonicalize} +566          1800         LR        GPR0,GPR0 <<< *Unreachable*
0x4962d320 {java/io/UnixFileSystem.canonicalize} +568          0000         Invalid Instruction
0x4962d322 {java/io/UnixFileSystem.canonicalize} +570          0000         Invalid Instruction
0x4962d324 {java/io/UnixFileSystem.canonicalize} +572          0100         Invalid Instruction
0x4962d326 {java/io/UnixFileSystem.canonicalize} +574          0000         Invalid Instruction
0x4962d328 {java/io/UnixFileSystem.canonicalize} +576          5040D51C     ST        GPR4,0x51C(,GPR13) <<< +562

+ (AOT load) java/lang/Class.desiredAssertionStatus()Z @ 0x4963EA34-0x4963F104 Q_SZ=0 Q_SZI=0 QW=1 j9m=49568C14 bcsz=20 compThread=0

0x4963eac6 {java/lang/Class.desiredAssertionStatus} +146          0D76         BASR      GPR7,GPR6
0x4963eac8 {java/lang/Class.desiredAssertionStatus} +148          47000001     NOP       0x1
0x4963eacc {java/lang/Class.desiredAssertionStatus} +152          A7F40006     J         *0xC U>> +164
0x4963ead0 {java/lang/Class.desiredAssertionStatus} +156          0000         Invalid Instruction
0x4963ead2 {java/lang/Class.desiredAssertionStatus} +158          0000         Invalid Instruction
0x4963ead4 {java/lang/Class.desiredAssertionStatus} +160          0100         Invalid Instruction
0x4963ead6 {java/lang/Class.desiredAssertionStatus} +162          0000         Invalid Instruction
0x4963ead8 {java/lang/Class.desiredAssertionStatus} +164          5040D51C     ST        GPR4,0x51C(,GPR13) <<< +152

+ (AOT load) java/lang/Class.getPrimitiveClass(Ljava/lang/String;)Ljava/lang/Class; @ 0x49603114-0x49605AB4 Q_SZ=0 Q_SZI=0 QW=1 j9m=49568C04 bcsz=222 compThread=0

0x49603348 {java/lang/Class.getPrimitiveClass} +564          0D76         BASR      GPR7,GPR6
0x4960334a {java/lang/Class.getPrimitiveClass} +566          47000002     NOP       0x2
0x4960334e {java/lang/Class.getPrimitiveClass} +570          A7F40009     J         *0x12 U>> +588
0x49603352 {java/lang/Class.getPrimitiveClass} +574          C00400000000 JLNOP     *0x0 U>> +574  <<< +574
0x49603358 {java/lang/Class.getPrimitiveClass} +580          0000         Invalid Instruction
0x4960335a {java/lang/Class.getPrimitiveClass} +582          0000         Invalid Instruction
0x4960335c {java/lang/Class.getPrimitiveClass} +584          0100         Invalid Instruction
0x4960335e {java/lang/Class.getPrimitiveClass} +586          0000         Invalid Instruction
0x49603360 {java/lang/Class.getPrimitiveClass} +588          5040D51C     ST        GPR4,0x51C(,GPR13) <<< +570
```

**Comment by joransiu** [comment]
Thanks @fjeremic for confirming!  Your changes LGTM.

**Comment by fjeremic** [comment]
@0xdaryl / @Leonardo2718 do either of you want to be committers for this one?

**Comment by 0xdaryl** [comment]
I arm wrestled @Leonardo2718 and I will be the committer on this one.

File: compiler/z/codegen/OMRInstOpCodeProperties.hpp [hunk 2]
```diff
@@ -4886,7 +4886,7 @@
    {
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRInstOpCodeProperties.hpp]: Could you update the copyright header in this file please?

File: compiler/z/CMakeLists.txt [hunk 3]
```diff
@@ -57,7 +57,6 @@ compiler_library(z
 	${CMAKE_CURRENT_LIST_DIR}/codegen/OMRCodeGenerator.cpp
```
Comment (0xdaryl) [inline compiler/z/CMakeLists.txt]: Could you update the copyright header in this file please?

File: compiler/z/codegen/OMRInstructionKindEnum.hpp [hunk 4]
```diff
@@ -103,3 +103,4 @@ IsI,               ///< is I instruction, i.e. SVC
 IsIE,              ///< is IE instruction, i.e. NIAI
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRInstructionKindEnum.hpp]: Could you update the copyright header in this file please?

File: compiler/z/codegen/S390Debug.cpp [hunk 5]
```diff
@@ -280,6 +280,9 @@ TR_Debug::printz(TR::FILE *pOutFile, TR::Instruction * instr)
       case TR::Instruction::IsNOP:
```
Comment (0xdaryl) [inline compiler/z/codegen/S390Debug.cpp]: Could you update the copyright header in this file please?

File: compiler/z/codegen/S390GenerateInstructions.hpp [hunk 6]
```diff
@@ -1522,4 +1522,29 @@ void generateShiftAndKeepSelected31Bit(
       int aToBit, int aShiftAmount, bool aClearOtherBits, bool aSetConditionCode);
```
Comment (0xdaryl) [inline compiler/z/codegen/S390GenerateInstructions.hpp]: Could you update the copyright header in this file please?

File: compiler/z/codegen/SystemLinkagezOS.hpp [hunk 7]
```diff
@@ -161,6 +161,8 @@ class S390zOSSystemLinkage : public TR::SystemLinkage
```
Comment (0xdaryl) [inline compiler/z/codegen/SystemLinkagezOS.hpp]: Could you update the copyright header in this file please?

File: compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp [hunk 8]
```diff
@@ -31,179 +31,6 @@
 #include "il/Node.hpp"
```
Comment (0xdaryl) [inline compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp]: Could you update the copyright header in this file please?

File: compiler/z/codegen/SystemLinkagezOS.hpp (hunk 7 above)
Comment (0xdaryl) [inline compiler/z/codegen/SystemLinkagezOS.hpp]: Could you update the copyright header in this file please?

File: compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp [hunk 9]
```diff
@@ -31,179 +31,6 @@
...
 #include "OMR/Bytes.hpp"
```
Comment (0xdaryl) [inline compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp]: Could you update the copyright header in this file please?

File: jitbuilder/build/files/target/z.mk [hunk 10]
```diff
@@ -61,5 +61,4 @@ JIT_PRODUCT_BACKEND_SOURCES+=\
     $(JIT_PRODUCT_DIR)/z/codegen/JBCodeGenerator.cpp \
```
Comment (0xdaryl) [inline jitbuilder/build/files/target/z.mk]: Could you update the copyright header in this file please?

File: fvtest/compilertest/build/files/target/z.mk [hunk 11]
```diff
@@ -58,7 +58,6 @@ JIT_PRODUCT_BACKEND_SOURCES+=\
     $(JIT_OMR_DIRTY_DIR)/z/codegen/snippet/PPA1Snippet.cpp \
```
Comment (0xdaryl) [inline fvtest/compilertest/build/files/target/z.mk]: Could you update the copyright header in this file please?

File: compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.hpp [hunk 12]
```diff
@@ -63,8 +63,6 @@ class XPLINKCallDescriptorSnippet : public TR::S390ConstantDataSnippet
     */
```
Comment (0xdaryl) [inline compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.hpp]: Could you update the copyright header in this file please?

File: compiler/z/codegen/S390GenerateInstructions.cpp [hunk 13]
```diff
@@ -3211,3 +3211,11 @@ template TR::Instruction * generateS390CompareAndBranchInstruction<int64_t>(
...
                     TR::Instruction        *preced = 0,
                     TR::RegisterDependencyConditions *cond = 0); 
+
+TR::Instruction*
+generateAlignmentNopInstruction(TR::CodeGenerator *cg, TR::Node *node, uint32_t alignment, TR::Instruction *preced)
+   {
+   if (preced)
+      return new (INSN_HEAP) TR::S390AlignmentNopInstruction(node, alignment, preced, cg);
+   return new (INSN_HEAP) TR::S390AlignmentNopInstruction(node, alignment, cg);
+   }
```
Comment (0xdaryl) [inline compiler/z/codegen/S390GenerateInstructions.cpp]: Weird EOL.

File: compiler/z/codegen/S390GenerateInstructions.hpp [hunk 14]
```diff
@@ -1522,4 +1522,29 @@ void generateShiftAndKeepSelected31Bit(
...
+ *     an appropriate number of NOP instructions at binary encoding time to satisfy the specified alignment.
+ *
+ *  \param cg
+ *     The code generator used to generate the instructions.
+ *
+ *  \param node
+ *     The node with which the generated instruction will be associated.
+ *
+ *  \param alignment
+ *     The algnment at which the next instruction in the instruction stream will be encoded at. For example specifying
```
Comment (0xdaryl) [inline compiler/z/codegen/S390GenerateInstructions.hpp]: typo:   alignment

File: compiler/z/codegen/SystemLinkagezOS.cpp [hunk 15]
```diff
@@ -615,15 +614,43 @@ TR::S390zOSSystemLinkage::genCallNOPAndDescriptor(TR::Instruction* cursor, TR::N
...
+      // 0x04 +----------------------------------+----------------------------------+----------------------------------+----------------------------------+
+      //      | Linkage and Return Value Adjust  | Parameter Adjust                                                                                       |
+      //      +----------------------------------+----------------------------------+----------------------------------+----------------------------------+
+      //
+      // We generate the XPLINK call descriptor inline right after the call instead of the literal pool because some
+      // z/OS 31-bit programs tend to be rather large, and the distance between the call location and the literal pool
+      // may exceed the number of bits we have to encode the offset in the NOP descriptor.
+
+      TR::LabelSymbol* xplinkCallDesciptorBeginLabel = generateLabelSymbol(cg());
+      TR::LabelSymbol* xplinkCallDesciptorEndLabel = generateLabelSymbol(cg());
```
Comment (0xdaryl) [inline compiler/z/codegen/SystemLinkagezOS.cpp]: Fix spelling of "Descriptor" in these variable names.  (unless that was intentional?)

File: compiler/z/codegen/SystemLinkagezOS.cpp [hunk 16]
```diff
@@ -636,6 +663,180 @@ TR::S390zOSSystemLinkage::genCallNOPAndDescriptor(TR::Instruction* cursor, TR::N
...
+         // For FastJNI, certain calls do not have to pass in receiver object.
+         if (cg()->fej9()->jniDoNotPassReceiver(resolvedMethod))
+            parmAreaOffset -= sizeof(uintptrj_t);
+         }
+#endif
+
+      uint32_t parmDescriptorFields = 0;
+
+
+      // WCode only logic follows for float parameter description fields
```
Comment (0xdaryl) [inline compiler/z/codegen/SystemLinkagezOS.cpp]: Does this comment imply the following code is no longer relevant?  i.e., is it WCode-specific?

If still relevant, perhaps the comment should be removed.  I imagine the "WCode" now means "system linkage".

File: compiler/z/codegen/OMRInstOpCodeProperties.hpp (hunk 2 above)
Comment (fjeremic) [reply compiler/z/codegen/OMRInstOpCodeProperties.hpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/CMakeLists.txt (hunk 3 above)
Comment (fjeremic) [reply compiler/z/CMakeLists.txt]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/SystemLinkagezOS.cpp (hunk 16 above)
Comment (fjeremic) [reply compiler/z/codegen/SystemLinkagezOS.cpp]: It is not relevant. I copied the code over from previous code. It is not "WCode" related. I've removed the line. Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/SystemLinkagezOS.cpp (hunk 15 above)
Comment (fjeremic) [reply compiler/z/codegen/SystemLinkagezOS.cpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/S390GenerateInstructions.hpp (hunk 14 above)
Comment (fjeremic) [reply compiler/z/codegen/S390GenerateInstructions.hpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/S390GenerateInstructions.cpp (hunk 13 above)
Comment (fjeremic) [reply compiler/z/codegen/S390GenerateInstructions.cpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.hpp (hunk 12 above)
Comment (fjeremic) [reply compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.hpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: fvtest/compilertest/build/files/target/z.mk (hunk 11 above)
Comment (fjeremic) [reply fvtest/compilertest/build/files/target/z.mk]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: jitbuilder/build/files/target/z.mk (hunk 10 above)
Comment (fjeremic) [reply jitbuilder/build/files/target/z.mk]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp (hunk 9 above)
Comment (fjeremic) [reply compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/SystemLinkagezOS.hpp (hunk 7 above)
Comment (fjeremic) [reply compiler/z/codegen/SystemLinkagezOS.hpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp (hunk 8 above)
Comment (fjeremic) [reply compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/SystemLinkagezOS.hpp (hunk 7 above)
Comment (fjeremic) [reply compiler/z/codegen/SystemLinkagezOS.hpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/S390GenerateInstructions.hpp (hunk 6 above)
Comment (fjeremic) [reply compiler/z/codegen/S390GenerateInstructions.hpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/S390Debug.cpp (hunk 5 above)
Comment (fjeremic) [reply compiler/z/codegen/S390Debug.cpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

File: compiler/z/codegen/OMRInstructionKindEnum.hpp (hunk 4 above)
Comment (fjeremic) [reply compiler/z/codegen/OMRInstructionKindEnum.hpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).

**Comment by fjeremic** [comment]
I'll have to rebase to fix the conflict. It was due to the `uintptrj_t` change. I'll wait until reviews are done to do this so as to avoid force pushing a change which will make the links to the current force pushes invalid.

**Comment by 0xdaryl** [comment]
Re-run CI after the merge conflicts are resolved.

**Comment by fjeremic** [comment]
@genie-omr build all

**Pull request merged**